### PR TITLE
Additional SSL Improvements

### PR DESCRIPTION
**Pull request by meyerrj**
Attempting to work around SSL certificate chain issues with the smartmetertexas.com website.

This patch is a work-around that uses the currently known SSL certificate's fingerprint for smartmetertexas.com which will expire in October. 

More information on the findings can be found in this PR:  https://github.com/home-assistant/core/pull/54493

This is also associated with an ongoing effort to resolve this issue: https://github.com/home-assistant/core/issues/40348

### CODE REVIEW

**Comment by meyerrj** [comment]
I added some logic that will test the SSL connection and return the currently known fingerprint, and then compare with the hard coded value before proceeding. This should allow for the code to respond to a new certificate if/when it is replaced before its expiration date.

**Comment by meyerrj** [comment]
@grahamwetzler  Would you be willing to review these changes and let me know if everything looks good on your end?

If everything is good, I can update the Home Assistant implementation to use the SSL logic in the examples with a version bump. :)

File: examples/get_meter_reading.py [hunk 1]
```diff
@@ -19,9 +27,37 @@
...
+                caiIndx = authorityInfoAccess.find(caiKey)
+                if (caiIndx > -1):
+                    caiValue = authorityInfoAccess[caiIndx:]
+                    caIssuersURI = reIssuersURI.findall(caiValue)[0]
+
+    if (caIssuersURI != None):
+        with urllib.request.urlopen(caIssuersURI) as certReq:
+            certData = certReq.read()
+            context.load_verify_locations(cafile=certifi.where(), cadata = certData)
+
```
Comment (grahamwetzler) [inline examples/get_meter_reading.py:57]: Can this be moved to another file and imported here? I’d like to keep the examples simple if possible.

File: examples/get_meters.py [hunk 2]
```diff
@@ -4,21 +4,62 @@
...
+        with urllib.request.urlopen(caIssuersURI) as certReq:
+            certData = certReq.read()
+            context.load_verify_locations(cafile=certifi.where(), cadata = certData)
+
+
+    # Re-enable checking
+    context.check_hostname = True
+    context.verify_mode = ssl.CERT_REQUIRED
+    context.options |= ssl.OP_NO_TLSv1 | ssl.OP_NO_TLSv1_1 | ssl.OP_NO_SSLv3 | ssl.OP_NO_SSLv2
+
```
Comment (grahamwetzler) [inline examples/get_meters.py:59]: Same as other example file

File: examples/get_meter_reading.py [hunk 3]
```diff
@@ -4,11 +4,19 @@
...
 import sys
-
 import aiohttp
 import pytz
+import ssl
+import socket
+import OpenSSL.crypto as crypto
+import certifi
+import re
+import urllib
```
Comment (grahamwetzler) [inline examples/get_meter_reading.py:14]: Please run `isort` to format the imports.

File: examples/get_meter_reading.py [hunk 4]
```diff
@@ -19,9 +27,37 @@
 
 
 async def main():
+
+    caiKey = 'CA Issuers - URI:'
```
Comment (grahamwetzler) [inline examples/get_meter_reading.py:31]: Please use `snake_case` for variables.

File: smart_meter_texas/__init__.py [hunk 5]
```diff
@@ -4,8 +4,11 @@
 import datetime
 import logging
 from random import randrange
+from aiohttp.http import RESPONSES
```
Comment (grahamwetzler) [inline smart_meter_texas/__init__.py:7]: This doesn’t seem to be needed

**Review by grahamwetzler — CHANGES_REQUESTED** [review]
Please see comments. Also please run `black` for consistent formatting. Thanks for your hard work!

**Comment by meyerrj** [comment]
@grahamwetzler I believe I've addressed the change requests in the latest commit. Please let me know if it looks good now. 😀

**Review by grahamwetzler — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**